back/app/controllers/omniauth_callback_controller.rb

```diff
@@ -81,6 +81,7 @@ def auth_callback(verify: false, authver_method: nil)
         ActiveRecord::Base.transaction do
           SideFxInviteService.new.before_accept @invite
           @user.save!
+          SideFxUserService.new.after_update(@user, nil)
```

Comment:
Don't think you need this as you're now logging a job in the `SideFxInviteService.after_accept` method. Also that method does other things that overlap with `after_update`

Reply:
As I mentioned in the description: We actually need this call to `after_update`, as the registration_completed_at is set on the fist save, (`@user.save!`) and we need to catch it immediately after the user is saved. the `after_accept` needs to happen after the invite is saved (which seem to also result in another user save, as at that point `registration_completed_at_previously_changed?` is false, and thus `after_accept` will not log the `registration_completed` activity).

This is only the case for the SSO flow. For non SSO invite acceptance the after_accept works fine, and I haven't inserted `after_update` in that flow.

I don't think my approach causes any unwanted side-effects and the performance hit of the occasional queueing of 2 jobs is not a performance concern.

Reply:
Also, it makes a kind of sense. Whenever a controller action causes a user to be saved... we call `after_update` immediately after the save.

Reply:
I've added comments to the 2 lines explain the reasoning.

Reply:
OK, makes sense now. The comment will help. I did read your description and then promptly forgot that reasoning when looking at the code. Also I did wonder why the tests weren't picking this up!